submission/programs/adsl.R

```diff
@@ -0,0 +1,455 @@
+  dplyr::select(-ZZZ)
+
+adsl03 <- adsl02 %>%
+  dplyr::left_join(sv00, by = c("STUDYID", "USUBJID")) 
+
+# Disposition -------------------------------------------------------------
+
+format_dcsreas <- function(dsdecod) {
+  dplyr::case_when(
+    dsdecod == "ADVERSE EVENT" ~ "Adverse Event",
+    dsdecod == "STUDY TERMINATED BY SPONSOR" ~ "Sponsor Decision",
+    dsdecod == "DEATH" ~ "Death",
+    dsdecod == "WITHDRAWAL BY SUBJECT" ~ "Withdrew Consent",
+    dsdecod == "PHYSICIAN DECISION" ~ "Physician Decision",
+    dsdecod == "PROTOCOL VIOLATION" ~ "Protocol Violation",
+    dsdecod == "LOST TO FOLLOW-UP" ~ "Lost to Follow-up",
+    dsdecod == "LACK OF EFFICACY" ~ "Lack of Efficacy"
+  )
+}
+
+adsl04 <- adsl03 %>%
+  dplyr::left_join(ds00, by = c("STUDYID", "USUBJID")) %>%
+  dplyr::select(-DSDECOD) %>%
+  admiral::derive_var_disposition_status(
+    dataset_ds = ds00,
+    new_var = EOSSTT,
+    status_var = DSDECOD, #this variable is removed after reformat
+    filter_ds = !is.na(USUBJID)
+  ) %>%
+  admiral::derive_vars_disposition_reason(
+    dataset_ds = ds00,
+    new_var = DCSREAS,
+    reason_var = DSDECOD,
+    filter_ds = !is.na(USUBJID),
+    format_new_vars = format_dcsreas #could not include dsterm in formatting logic
+  ) %>%
+  dplyr::mutate(DCSREAS = ifelse(DSTERM == "PROTOCOL ENTRY CRITERIA NOT MET", "I/E Not Met", DCSREAS))
+
+# Baseline variables ------------------------------------------------------
+# selection definition from define
+
+vs00 <- vs %>%
+  dplyr::filter((VSTESTCD == "HEIGHT" & VISITNUM == 1) | (VSTESTCD == "WEIGHT" & VISITNUM == 3)) %>%
+  dplyr::mutate(AVAL = round(VSSTRESN, digits = 1)) %>%
+  dplyr::select(STUDYID, USUBJID, VSTESTCD, AVAL) %>%
+  tidyr::pivot_wider(names_from = VSTESTCD, values_from = AVAL, names_glue = "{VSTESTCD}BL") %>%
+  dplyr::mutate(BMIBL = round(WEIGHTBL/(HEIGHTBL/100)^2, digits = 1),
+                BMIBLGR1 = format_bmi(BMIBL)
+               ) 
+
+sc00 <- sc %>%
+  dplyr::filter(SCTESTCD == "EDLEVEL") %>%
+  dplyr::select(STUDYID, USUBJID, SCTESTCD, SCSTRESN) %>%
+  tidyr::pivot_wider(names_from = SCTESTCD, values_from = SCSTRESN, names_glue = "EDUCLVL")
+
+adsl05 <- adsl04 %>%
+  dplyr::left_join(vs00, by = c("STUDYID", "USUBJID")) %>%
+  dplyr::left_join(sc00, by = c("STUDYID", "USUBJID"))
+
+# Disease information -----------------------------------------------------
+
+visit1dt <- sv %>%
+  dplyr::filter(VISITNUM == 1) %>%
+  admiral::derive_vars_dt(
+    dtc = SVSTDTC,
+    new_vars_prefix = "VISIT1",
+  ) %>%
+  dplyr::select(STUDYID, USUBJID, VISIT1DT)
+
+visnumen <- sv %>%
+  dplyr::filter(VISITNUM < 100) %>%
+  dplyr::arrange(STUDYID, USUBJID, SVSTDTC) %>%
+  dplyr::group_by(STUDYID, USUBJID) %>%
+  dplyr::slice(n()) %>%
+  dplyr::ungroup() %>%
+  dplyr::mutate(VISNUMEN = ifelse(round(VISITNUM, digits = 0) == 13, 12, round(VISITNUM, digits = 0))) %>%
+  dplyr::select(STUDYID, USUBJID, VISNUMEN)
```

Comment:
According to the SAP, this derivation should be done using DS:
![image](https://user-images.githubusercontent.com/113363343/205073493-6a0233b1-4080-492d-9317-9c6e92b1ef7e.png)

Reply:
This is the define you show. I saw but this information is not in DS. I used DS.DSSTDTC information which should be sufficient.